src/codec_svt.c

```diff
@@ -205,6 +205,11 @@ static avifResult svtCodecEncodeImage(avifCodec * codec,
         }
 #endif
 
+#if SVT_AV1_CHECK_VERSION(3, 0, 0)
+        svt_config->lossless = quantizer == AVIF_QUANTIZER_LOSSLESS;
```

Comment:
apps/avifenc.c has the following code:

```
    // Check global lossless parameters and set to default if needed.
    if (lossless) {
        ...
        // Codec.
        const char * codecName = avifCodecName(settings.codecChoice, AVIF_CODEC_FLAG_CAN_ENCODE);
        ...
        } else if (codecName && !strcmp(codecName, "svt")) {
            fprintf(stderr, "SVT-AV1 doesn't support lossless encoding yet: https://gitlab.com/AOMediaCodec/SVT-AV1/-/issues/1636\n");
            goto cleanup;
        }
```

We should update that check with an SVT-AV1 version check. I am not sure how to get the SVT-AV1 version number in apps/avifenc.c ...

Reply:
Done.

Reply:
apps/avifenc.c is not modified in this PR, so this is not done.

It is fine to just add a TODO comment.

Reply:
Sorry, I discussed checking the SVT version in `avifsvttest.cc` with Vincent and thought your comment was the same thing.

Reply:
I went with checking values returned by `avifCodecVersions()` in `avifenc.c` and by `avifCodecVersionSvt()` in `avifsvttest.c` rather than adapting CMakeLists.txt or the libavif public API.

Reply:
What you implemented is what I would have done. But James suggested we move the version check into src/codec_svt.c and let the encoding fail rather than produce a lossy output incorrectly. I am sorry I forgot to post James's suggestion last night.

I will review your solution first today.

Reply:
We thought about that. Unfortunately calling libavif with SVT at `AVIF_QUALITY_BEST` should not result in a failure, right? Or we have to change the value of `AVIF_QUALITY_LOSSLESS` so that it differs from `AVIF_QUALITY_BEST` etc.

Reply:
> We thought about that. Unfortunately calling libavif with SVT at `AVIF_QUALITY_BEST` should not result in a failure, right? Or we have to change the value of `AVIF_QUALITY_LOSSLESS` so that it differs from `AVIF_QUALITY_BEST` etc.

That's unfortunate. I guess this may be the only option then.

Reply:
libavif quality 100 (or equivalently quantizer 0) means lossless. So `AVIF_QUALITY_LOSSLESS` is just a synomym for `AVIF_QUALITY_BEST`.

You can see this in the `avifenc` help message:

```
    -q,--qcolor Q                     : Quality for color in 0..100 where 100 is lossless
    --qalpha Q                        : Quality for alpha in 0..100 where 100 is lossless
    ...
    --qgain-map Q                     : Quality for the gain map in 0..100 where 100 is lossless
```

If libavif is called with SVT-AV1 v2.3.0 at `AVIF_QUALITY_BEST`, a failure is expected. Producing a lossily-encoded output is arguably a bug.

Reply:
The API does not mention lossless:

https://github.com/AOMediaCodec/libavif/blob/bdba09d6429c0f19dbeb487d8042ab6fb4a2d868/include/avif/avif.h#L1541-L1542

`avifenc -c svt -q 100` could be forbidden with SVT < 3.0.0 indeed.

Reply:
We can move the definition of the `AVIF_QUALITY_LOSSLESS` macro around to make it clear that it is the same as `AVIF_QUALITY_BEST`: https://github.com/AOMediaCodec/libavif/pull/2638